app/presenters/content_item/links_out.rb

```diff
@@ -0,0 +1,107 @@
+module ContentItem
+  module LinksOut
+    def links_out_supergroups
+      @links_out_supergroups ||= fetch_links_out_supergroups
+    end
+
+    def fetch_links_out_supergroups
+      links_out.map { |link| link["supergroup"] }.uniq
+    end
+
+    def links_out_subgroups
+      @links_out_subgroups ||= fetch_links_out_subgroups
+    end
+
+    def fetch_links_out_subgroups
+      subgroups = []
+      links_out.each do |link|
+        if link["type"] == "content_purpose_subgroup"
+          subgroups << link["title"]
+        end
+      end
+      subgroups.uniq
+    end
+
+    def links_out
+      @links_out ||= fetch_links_out
+    end
+
+    def fetch_links_out
+      document_type_link_rule = DocumentTypeLinkRule.new(content_item)
+      return document_type_link_rule.links if document_type_link_rule.any?
+      subgroup_link_rule = SubgroupLinkRule.new(content_item)
+      return subgroup_link_rule.links if subgroup_link_rule.any?
+      SupergroupLinkRule.new(content_item).links
+    end
+
+    class LinkRule
+      def initialize(content_item)
+        @content_item = content_item
+        @rules = load_rules
+        @links = fetch_links
+      end
+
+      def any?
+        !@links.nil?
```

Comment:
Should we use the public `links` interface here since it implements a default?

We could maybe delegate the `any?` method to the links object and initialise the default like so:

```rb
class LinkRule
  extend Forwardable
  
  def initialize(content_item)
    @content_item = content_item
    @rules = load_rules
    @links = fetch_links || []
  end
  
  def_delegator :@links, :any?

  # ...
end
```